### PR TITLE
Aro 2019-09-30 preview API Version

### PR DESCRIPTION
**Pull request by JackQuincy**
Adding swagger spec for the newest API Version for ARO.
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30-preview/openShiftManagedClusters.json [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+        "subnetCidr": {
+          "type": "string",
+          "description": "Subnet CIDR for the peering."
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        }
+      },
+      "required": [
```
Comment (raych1) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30-preview/openShiftManagedClusters.json]: What's the difference between this Profile and `OpenShiftManagedClusterAgentPoolProfile` which have `name` as required property but here doesn't?

**Comment by raych1** [comment]
@JackQuincy, please copy the previous version API swagger as first commit then work on top of it for the new change when adding new API version. It will be easier for the review on the new change. Another thing, it's better to have the folder name called as `...nerService/preview/2019-09-30/...` without literal `preview` after API version so that it will be easier to move the exact whole folder to stable version.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30-preview/openShiftManagedClusters.json (hunk 1 above)
Comment (JackQuincy) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30-preview/openShiftManagedClusters.json]: The master pool is only one and can easily be given a default name. The Agent pools are 1+ and need names to differentiate them.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/openShiftManagedClusters.json [hunk 2]
```diff
@@ -0,0 +1,817 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerServiceClient",
+    "description": "The Container Service Client.",
+    "version": "2019-04-30"
```
Comment (KrisBash) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/openShiftManagedClusters.json]: Version does not match folder name or PR title.

**Review by KrisBash — COMMENTED** [review]
I compared with 2018-09-30 preview, and the actual API changes seem fine. I think the only required fix is the version property in the swagger spec

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/openShiftManagedClusters.json (hunk 2 above)
Comment (JackQuincy) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/openShiftManagedClusters.json]: Thanks. I'll fix this soon

**Comment by JackQuincy** [comment]
Internally we have decided to add an enabled disabled flag here as well so that we can allow the user to remove the logging integration. And that change is waiting on our partner team in redhat. Once that is finalized I'll update this pr. Should be by EOW

**Comment by JackQuincy** [comment]
took a little longer then expected but this is now updated with the final schema for this version and ready for review

**Comment by JackQuincy** [comment]
@raych1 this is ready for review again sorry for the long time between revisions

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/examples/OpenShiftManagedClustersGet.json [hunk 3]
```diff
@@ -69,7 +69,8 @@
             ]
           },
           "monitorProfile": {
-            "workspaceResourceID": "/subscriptions/subid1/resourcegroups/rg1/providers/Microsoft.OperationalInsights/workspaces/workspacename1"
+          "enabled": true,
```
Comment (raych1) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30/examples/OpenShiftManagedClustersGet.json]: Please correct indent.

File: specification/containerservice/resource-manager/readme.python.md [hunk 4]
```diff
@@ -126,3 +127,14 @@ python:
   namespace: azure.mgmt.containerservice.v2017_07_01
   output-folder: $(python-sdks-folder)/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2017_07_01
 ```
+
+### Tag: package-2019-09-preview-only and python
+
+These settings apply only when `--tag=package-2019-08-preview-only --python` is specified on the command line.
```
Comment (raych1) [inline specification/containerservice/resource-manager/readme.python.md]: Typo?

**Comment by raych1** [comment]
@KrisBash , please review the changes, thanks.

**Review by KrisBash — APPROVED** [review]
LGTM

File: specification/containerservice/resource-manager/readme.python.md (hunk 4 above)
Comment (JackQuincy) [reply specification/containerservice/resource-manager/readme.python.md]: copy paste error. fixing

**Comment by JackQuincy** [comment]
@raych1 this should be ready for review again. Sorry for slow turn around.

**Review by raych1 — APPROVED** [review]

**Pull request merged**